presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java

```diff
@@ -42,8 +46,33 @@
 
 public abstract class BaseTestJdbcResultSet
 {
+    private static final Logger log = Logger.get(BaseTestJdbcResultSet.class);
+
+    private final Supplier<String> prestoServerVersion = Suppliers.memoize(this::determinePrestoVersion);
+
     protected abstract Connection createConnection() throws SQLException;
 
+    @SuppressWarnings("JDBCResourceOpenedButNotSafelyClosed")
+    @BeforeMethod
+    public void setup()
+    {
+        synchronized (this) {
+            determinePrestoVersion();
+        }
+    }
+
+    private String determinePrestoVersion()
+    {
+        try (StatementWrapper statementWrapper = new StatementWrapper()) {
+            ResultSet rs = statementWrapper.getStatement().executeQuery("SELECT node_version FROM system.runtime.nodes");
```

Comment:
node_version -> DISTINCT node_version

plus `assertFalse(rs.next());` at the end (exactly 1 row)